content/200-concepts/100-components/01-prisma-schema/07-referential-actions.mdx

```diff
@@ -0,0 +1,327 @@
+  posts Post[]
+}
+```
+
+The foreign key ensures that when saving rows into the `Post` table, any value in the `authorId` [scalar field](./data-model#scalar-fields) must be referencing an existing value in the `id` column of the `User` table. 
+
+When a [migration](../prisma-migrate) is run the following SQL is produced:
+
+```sql
+CREATE TABLE "Post" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    "title" TEXT NOT NULL,
+    "authorId" INTEGER NOT NULL,
+    FOREIGN KEY ("authorId") REFERENCES "User" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+);
+
+CREATE TABLE "User" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT
+);
+```
+
+And the foreign key relation that was created can be seen here:
+
+```sql
+FOREIGN KEY ("authorId") REFERENCES "User" ("id") ON DELETE RESTRICT ON UPDATE CASCADE
+```
+
+### Current behavior 
+
+Prisma creates foreign keys for each relation between data models, with the following defaults: 
+
+- `ON DELETE CASCADE` - The users posts are deleted when the user is deleted.
+- `ON UPDATE CASCADE` - When a users `id` is updated, the corresponding `authorId` on the `Posts` table is updated too.
+
+When invoking the [`delete()`](../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../prisma-client/crud#delete-all-records) methods using the Prisma Client, a runtime check is performed and the deletion of records prevented if they are referencing related objects. **This prevents cascade behavior**. 
+
+If you need to replicate the cascade behavior, you _can_ use [`$raw`](../prisma-client/raw-database-access) SQL queries to [delete multiple referenced records](../prisma-client/crud#deleting-all-data-with-raw-sql--truncate) (such as users blog posts) and the deletion will be successful, this is because the Prisma Client will **not** perform runtime checks.
+
+### New behavior
+
+With the feature flag **enabled**, the default value of `ON DELETE` is [`Restrict`](#restrict) (`ON DELETE RESTRICT`).
+
+The Prisma Client will **not** perform any runtime checks before deleting related records with `delete()` or `deleteAll()`. This means the deletion will succeed or fail according to the underlying foreign key relations and any referential actions set. 
+
+By default the deletion will be prevented by the database because of the **new** default action, [`Restrict`](#restrict) (`ON DELETE RESTRICT`). To make it succeed you will need to manually add the [`Cascade`](#cascade) action. 
+
+#### Required relations
+
+| Clause      |  < 2.26.0       | 2.26.0 +   |
+| :---------- | :-------------- | :--------- |
+| `onDelete`  | `Cascade`       | `Restrict` |
+| `onUpdate`  | `Cascade`       | `Cascade`  |
+
+#### Optional relations
+
+| Clause      |  < 2.26.0       | 2.26.0 +   |
+| :---------- | :-------------- | :--------- |
+| `onDelete`  | `SetNull`       | `SetNull`  |
+| `onUpdate`  | `Cascade`       | `Cascade`  |
```

Comment:
Why are they under the "New" headline if they show a comparison of before and after? I think they might require an additional headline `Comparison between referential actions default, before and after` or similar maybe?

Reply:
I was worried this wouldn't be clear that they are separate. Will add a new headline for clarity